cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -54,6 +79,9 @@ type InitConfiguration struct {
 	// CertificateKey sets the key with which certificates and keys are encrypted prior to being uploaded in
 	// a secret in the cluster during the uploadcerts init phase.
 	CertificateKey string
+
+	// PublicKeyAlgorithm sets the algorithm used for generating encryption keys: RSA or ECDSA.
+	PublicKeyAlgorithm PublicKeyAlgorithm
```

Comment:
InitConfiguration vs something else is an interesting topic.
do we gain anything if this is part of ClusterConfiguration?

i'm hoping that nobody would like to switch algorithms on a working cluster, but unfortunately people will want to do that on their existing RSA based clusters. this implies CA rotation, so it opens a whole can of worms.

i'm starting to think that this feature needed a KEP (enhancement proposal that clarifies all the details).

Reply:
It makes more sense to me to have this field in the ClusterConfiguration. I am too concerned if certificate renewal will work as expected when switching from RSA to ECDSA.